README.md

```diff
@@ -1 +1,101 @@
+
+## IO ports
+The input and output ports of the Post-Processing SIMD accelerator are shown in the table below.
+
+The Post-Processing SIMD accelerator uses a simplified CSR request/response interface for CSR write/read operation. A more detailed description of the CSR operation interface can be found at [here](https://kuleuven-micas.github.io/snitch_cluster/rm/snax_cluster.html).
+
+The Post-Processing SIMD accelerator uses the Decoupled interface for input and output data. A more detailed description of the Decoupled interface can be found at [here](https://www.chisel-lang.org/docs/explanations/interfaces-and-connections#the-standard-ready-valid-interface-readyvalidio--decoupled).
+
+|Signal bundle| Signals | Signal name in generated SV | Width | Dir | Description |
+| - | - | - | - | - | - |
+| csr.req | data | io_csr_req_bits_data | 32| In| The write data from CSR request |
+|  | addr | io_csr_req_bits_addr | 32| In| The address indicating which CSR to be wrote or read |
+|  | write | io_csr_req_bits_write | 1 | In| The signal indicates this request is for CSR write or read |
+|  | valid | io_csr_req_valid | 1 | In| The signal indicates if this request is valid |
+|  | ready | io_csr_req_ready | 1 | Out| The signal indicates if the accelerator is ready for this CSR operation|
+| csr.rsp | data | io_csr_rsp_bits_data | 32| Out| The response data for CSR read operation |
+|  | valid | io_csr_rsp_valid | 1 | Out| The signal indicates if this response is valid |
+|  | ready | io_csr_rsp_ready | 1 | In| The signal indicates if the SNAX core is ready for this CSR response |
+| data.input_i | bits | io_data_input_i_bits | `laneLen * 32`| In| The input data content|
+|  | valid | io_data_input_i_valid | 1| In| The signal indicates if this input data is valid |
+|  | ready | io_data_input_i_ready | 1| Out| The signal indicates if the accelerator is ready for this input |
+| data.out_o | bits | io_data_out_o_bits | `laneLen * 8`| Out| The output data content |
+|  | valid | io_data_out_o_valid | 1| Out| The signal indicates if this output data is valid |
+|  | ready | io_data_out_o_ready | 1| In| The signal indicates if the SNAX core is ready for this output data|
+
+The  data.input_i.bits are spited into each PE and the results from each PE are gathered to the data.out_o.bits as indicated at the figure below. Each PE share the same control data from CSRs.
+
+<p align="center">
+  <img src="./docs/microarch_detail.svg" alt="">
+</p>
+
+## Functional description
+The Functional description in the mathematical formula of the Post-Processing SIMD Accelerator is defined as below. The Post-Processing-Func below is the [post-processing kernel](https://gist.github.com/jorendumoulin/83352a1e84501ec4a7b3790461fee2bf) for TinyML workload. Vu is the spatial unrolling factor which is indicated by `laneLen`.
+
+```
+for (ti = 0 to VEC_LEN/Vu – 1):
```

Comment:
I don't think you need to put the temporal loop here since the SIMD is a parallel process. You're not even configuring it here in the first place. OR if you want to make it clear, describe it as a parallel process first THEN you describe later how it can be used in a temporal loop.